### PR TITLE
Trimaran minor bug fixes

### LINKED ISSUE #613: [Trimaran] minor bug fixes
### Area

- [X] Scheduler
- [ ] Controller
- [ ] Helm Chart
- [ ] Documents

### Other components

_No response_

### What happened?

This issue fixes below minor bugs found during code walk through:
1. Bug-1: Trimaran collector.go is printing `the type of the value` but not `the value` below.
> https://github.com/kubernetes-sigs/scheduler-plugins/blob/efddd13fd45d3e35018976ba029f9980a3b77a18/pkg/trimaran/collector.go#L126
2. Bug-2: Trimaran resourcestats.go function GetNodeRequestsAndLimits() needs to call `SetMaxLimits()` in the if condition also. The `SetMaxLimits()` is currently called only in the else condition in the below if-else block.
> https://github.com/kubernetes-sigs/scheduler-plugins/blob/efddd13fd45d3e35018976ba029f9980a3b77a18/pkg/trimaran/resourcestats.go#L199
3. Bug-3: Trimaran resourcestats.go function GetNodeRequestsAndLimits() needs to call `setMin` on `nodeLimit` as well. Currently it is called for `nodeRequest` only.
> https://github.com/kubernetes-sigs/scheduler-plugins/blob/efddd13fd45d3e35018976ba029f9980a3b77a18/pkg/trimaran/resourcestats.go#L212
4. Added couple of Go unit test cases that exercise the above bug fixes.

### What did you expect to happen?

The issues above are functionality bugs.
1. If we don't fix the format specifier as part of the Bug-1 reported above, then the error statement will miss lead the user.
   > Function `TestNewCollectorSpecs` with test cases is added to scheduler-plugins/pkg/trimaran/collector_test.go to exercise this. The test cases in this function will fail without this source code change.
2. If we don't add this statement `SetMaxLimits(requested, limits)` for if statement to address the Bug-2 reported above, then user might falsely set `nodeLimit.MilliCPU` to some value that is greater than `nodeRequest.MilliCPU`.
   > `test-3` is added to the function TestGetNodeRequestsAndLimits() of scheduler-plugins/pkg/trimaran/resourcestats_test.go to exercise this. This test case will fail without this source code change.
3. If we don't add this statement `setMin(&nodeLimit.MilliCPU, capCpu)` to address the Bug-3 reported above, then user might falsely set `nodeLimit.MilliCPU` to some value that is greater than CPU capacity available on that node.
   > `test-2` is added to the function TestGetNodeRequestsAndLimits() of scheduler-plugins/pkg/trimaran/resourcestats_test.go to exercise this. This test case will fail without this source code change.

### How can we reproduce it (as minimally and precisely as possible)?

Test cases were added to reproduce these issues. Without the change to the files `scheduler-plugins/pkg/trimaran/collector.go` and `scheduler-plugins/pkg/trimaran/resourcestats.go`, the test cases that are added as specified above will fail.

### Anything else we need to know?

The test cases that are added are increasing the source code coverage percentage.

### Kubernetes version

<details>

```console
$ kubectl version
```

</details>

### Scheduler Plugins version

<details>
</details>

### PR DESCRIPTION
**Pull request by knarayan**
#### What type of PR is this?
/kind cleanup
#### What this PR does / why we need it:
Please refer to the issue #613
#### Which issue(s) this PR fixes:
Fixes #613

#### Does this PR introduce a user-facing change?

```release-note
Fixing minor bugs:
- correctly printing the value of a variable instead of the type of the value of a variable
- removing unnecessary intermediate variable assignment

```

### CODE REVIEW

**Comment by Huang-Wei** [comment]
@wangchen615 @atantawi could you take a look?

**Comment by atantawi** [comment]
@Huang-Wei Sure, I'll take a look.

**Comment by knarayan** [comment]
I request your review @zwpaper and @PiotrProkop on this PR. Thank you.

File: pkg/trimaran/resourcestats.go [hunk 1]
```diff
@@ -195,24 +195,24 @@ func GetNodeRequestsAndLimits(podInfosOnNode []*framework.PodInfo, node *v1.Node
...
 			limits = GetResourceLimits(p)
-			// make sure limits not less than requests
-			SetMaxLimits(requested, limits)
 		}
-		reqCpu := requested.MilliCPU
-		reqMem := requested.Memory
-		limitCpu := limits.MilliCPU
-		limitMem := limits.Memory
+		// make sure limits not less than requests
+		SetMaxLimits(requested, limits)
```
Comment (atantawi) [inline pkg/trimaran/resourcestats.go:200]: No need to take `SetMaxLimits()` outside loop as the last element in the list, namely `pod` had already passed that check in [CreatePodResourcesStateData()](https://github.com/kubernetes-sigs/scheduler-plugins/blob/6df71a420673837e2aaf90540f67d3cac18a799a/pkg/trimaran/lowriskovercommitment/lowriskovercommitment.go#L261).

File: pkg/trimaran/resourcestats.go [hunk 2]
```diff
@@ -195,24 +195,24 @@ func GetNodeRequestsAndLimits(podInfosOnNode []*framework.PodInfo, node *v1.Node
...
+		nodeLimit.MilliCPU += limits.MilliCPU
+		nodeLimit.Memory += limits.Memory
 	}
 	// cap requests by node capacity
 	setMin(&nodeRequest.MilliCPU, capCpu)
 	setMin(&nodeRequest.Memory, capMem)
 	setMin(&nodeRequestMinusPod.MilliCPU, capCpu)
 	setMin(&nodeRequestMinusPod.Memory, capMem)
+	// cap limits by node capacity
+	setMin(&nodeLimit.MilliCPU, capCpu)
```
Comment (atantawi) [inline pkg/trimaran/resourcestats.go:214]: No need to cap the values of limits to node capacities. It is valid for a user to specify a limit that is higher than node capacity. This case is handled in [computeRisk()](https://github.com/kubernetes-sigs/scheduler-plugins/blob/6df71a420673837e2aaf90540f67d3cac18a799a/pkg/trimaran/lowriskovercommitment/lowriskovercommitment.go#L206).

File: pkg/trimaran/resourcestats.go (hunk 1 above)
Comment (knarayan) [reply pkg/trimaran/resourcestats.go]: Sure @atantawi , reversed this change.

File: pkg/trimaran/resourcestats.go (hunk 2 above)
Comment (knarayan) [reply pkg/trimaran/resourcestats.go]: Sure @atantawi , reversed this change.

**Review by atantawi — CHANGES_REQUESTED** [review]
Please squash 6 commits into one.

**Review by atantawi — APPROVED** [review]

**Comment by PiotrProkop** [comment]
@knarayan is this PR intended to fix only Bug-1 from issue description or also Bug-2 and Bug-3?
Cause merging this PR will auto close corresponding issue. Either add fixed for Bug-2 and Bug-3 to this PR or create new issues to track them.

**Comment by knarayan** [comment]
@PiotrProkop this PR is intended to fix only Bug-1 and also the point 4 from issue #613 description.
I am not opening any new issue since Bug-2 and Bug-3 are not bugs as such as clarified in https://github.com/kubernetes-sigs/scheduler-plugins/issues/613#issuecomment-1685864101 

Hence, auto closing issue #613 by merging this PR is fine. Thank you.

**Comment by knarayan** [comment]
Request you @PiotrProkop and @zwpaper to review this PR. Thanks.

**Comment by zwpaper** [comment]
hi @knarayan，thanks for the contribution, it seems that the PR is reviewed and may be good to go, but we may need a release note for this bugfix, can you update the PR's description to add the release note.

And after that you can ask for approval by commenting: 

```
/assign @ffromani 
```

**Review by ffromani — COMMENTED** [review]
/approve

**Pull request merged**